iis-wf/iis-wf-ingest-pmc/src/test/java/eu/dnetlib/iis/wf/ingest/pmc/metadata/MetadataImporterMain.java

```diff
@@ -29,30 +32,52 @@ public class MetadataImporterMain {
     
     private static final String encoding = "utf-8"; 
     
+    private static final String parsedFileSuffix = ".parsed";
+    
     public static void main(String[] args) throws IOException, ParserConfigurationException, SAXException {
         if (args.length == 1) {
             if (StringUtils.isBlank(args[0])) {
                 throw new RuntimeException("No valid location provided!");
             }
             
-            ExtractedDocumentMetadata.Builder builder  = ExtractedDocumentMetadata.newBuilder();
-            builder.setId("");
-            builder.setText("");
-            extractMetadata(getFileContent(args[0]), builder);
-            
-            DatumWriter<ExtractedDocumentMetadata> writer = new SpecificDatumWriter<>(ExtractedDocumentMetadata.class);
-            ByteArrayOutputStream stream = new ByteArrayOutputStream();
-            Encoder jsonEncoder = EncoderFactory.get().jsonEncoder(ExtractedDocumentMetadata.getClassSchema(), stream);
-            writer.write(builder.build(), jsonEncoder);
-            jsonEncoder.flush();
-            System.out.println(new String(stream.toByteArray(), encoding));
+            File file = new File(args[0]);
+            if (file.isDirectory()) {
+                String[] fileLocs = file.list(new FilenameFilter() {
+                    
+                    @Override
+                    public boolean accept(File dir, String name) {
+                        return !name.endsWith(parsedFileSuffix);
+                    }
+                });
+                for (String fileLoc : fileLocs) {
+                    processFile(new File(file, fileLoc).getAbsolutePath());
+                }
+            } else {
+                processFile(args[0]);
+            }
             
         } else {
             throw new RuntimeException("invalid number of input arguments: " + args.length 
                     + ", expecting a single argument with JATS XML file location!");
         }
     }
     
+    private static void processFile(String fileLocation) throws ParserConfigurationException, SAXException, IOException {
+        ExtractedDocumentMetadata.Builder builder  = ExtractedDocumentMetadata.newBuilder();
+        builder.setId("");
+        builder.setText("");
+        extractMetadata(getFileContent(fileLocation), builder);
+        
+        DatumWriter<ExtractedDocumentMetadata> writer = new SpecificDatumWriter<>(ExtractedDocumentMetadata.class);
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        Encoder jsonEncoder = EncoderFactory.get().jsonEncoder(ExtractedDocumentMetadata.getClassSchema(), stream);
+        writer.write(builder.build(), jsonEncoder);
+        jsonEncoder.flush();
+        String fileContent = new String(stream.toByteArray(), encoding);
+        System.out.println(fileContent);
+        FileUtils.writeStringToFile(new File(fileLocation + parsedFileSuffix), fileContent, encoding);
```

Comment:
Possibly not worth other necessary changes, but we could avoid the `FileUtils` dependency by using standard `Files.write(Path, byte[])`

Reply:
As already discussed, it is a good idea but to get rid of this utility class and dependency we'd need to alter the code outside of this PR's scope. Since this PR does not introduce any new dependency, it just relies on the set of already available deps, we could do this optimization but as a part of a different task.